cardano-testnet/src/Testnet/Components/SPO.hs

```diff
@@ -0,0 +1,327 @@
+  GHC.withFrozenCallStack $ do
+    let tempAbsPath' = unTmpAbsPath tempAbsPath
+        testnetMag = cardanoTestnetMagic cTestnetOpts
+        oFpAbs = tempAbsPath' </> outputFp
+
+    sAddr <- case deserialiseAddress AsStakeAddress $ Text.pack stakeAddr of
+               Just sAddr -> return sAddr
+               Nothing -> H.failWithCustom GHC.callStack Nothing $ "Invalid stake address: " <> stakeAddr
+
+    void $ execCli' execConfig
+      [ "query", "stake-address-info"
+      , "--address", stakeAddr
+      , "--testnet-magic", show @Int testnetMag
+      , "--out-file", oFpAbs
+      ]
+
+    pledgerStakeInfo <- H.leftFailM $ H.readJsonFile oFpAbs
+    dag@(DelegationsAndRewards (rewardsMap, _delegMap)) <- H.noteShowM $ H.jsonErrorFail $ Aeson.fromJSON @DelegationsAndRewards pledgerStakeInfo
+    case Map.lookup sAddr rewardsMap of
+      Nothing -> H.failWithCustom GHC.callStack Nothing
+                   $ unlines [ "Stake address: "
+                             , Text.unpack (serialiseToBech32 sAddr)
+                             , "was not registered"
+                             , "Current registered stake keys: "
+                             , show $ map serialiseToBech32 $ Map.keys rewardsMap
+                             ]
+      Just _ -> return dag
+
+
+createStakeDelegationCertificate
+  :: (MonadTest m, MonadCatch m, MonadIO m, HasCallStack)
+  => TmpAbsolutePath
+  -> AnyCardanoEra
+  -> FilePath -- ^ Delegate stake verification key file
+  -> String -- ^ Pool id
+  -> FilePath
+  -> m ()
+createStakeDelegationCertificate tempAbsPath anyCera delegatorStakeVerKey poolId outputFp =
+  GHC.withFrozenCallStack $ do
+    let tempAbsPath' = unTmpAbsPath tempAbsPath
+    void $ execCli
+      [ "stake-address", "delegation-certificate"
+      , convertToEraFlag anyCera
+      , "--stake-verification-key-file", delegatorStakeVerKey
+      , "--stake-pool-id", poolId
+      , "--out-file", tempAbsPath' </> outputFp
+      ]
+
+createStakeKeyRegistrationCertificate
+  :: (MonadTest m, MonadCatch m, MonadIO m, HasCallStack)
+  => TmpAbsolutePath
+  -> AnyCardanoEra
+  -> FilePath -- ^ Stake verification key file
+  -> FilePath -- ^ Output file path
+  -> m ()
+createStakeKeyRegistrationCertificate tempAbsPath anyCEra stakeVerKey outputFp =
+  GHC.withFrozenCallStack $ do
+    let tempAbsPath' = unTmpAbsPath tempAbsPath
+
+    void $ execCli
+      [ "stake-address", "registration-certificate"
+      , convertToEraFlag anyCEra
+      , "--stake-verification-key-file", stakeVerKey
+      , "--out-file", tempAbsPath' </> outputFp
+      ]
+
+convertToEraFlag :: AnyCardanoEra -> String
+convertToEraFlag (AnyCardanoEra e) =
+  case e of
+    ConwayEra -> "--conway-era"
+    BabbageEra -> "--babbage-era"
+    AlonzoEra -> "--alonzo-era"
+    MaryEra -> "--mary-era"
+    AllegraEra -> "--allegra-era"
+    ShelleyEra -> "--shelley-era"
+    ByronEra -> "--byron-era"
```

Comment:
We'll be able to convert to the era-based transactions after the next integration with `cardano-api` and `cardano-cli` right?